projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib-checkout.migration.ts

```diff
@@ -0,0 +1,2679 @@
+        importPath: '@spartacus/product-configurator/rulebased',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'ActiveCartOrderEntriesContext',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ProductImportInfoService',
+        importPath: '@spartacus/cart/base/core',
+      },
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'AddedToCartDialogComponent',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'ModalService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'AddToCartComponent',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'ModalService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'CurrentProductService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'ChangeDetectorRef',
+        importPath: '@angular/core',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'CmsComponentData',
+        importPath: '@spartacus/storefront',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ModalService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+      {
+        className: 'EventService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'ProductListItemContext',
+        importPath: '@spartacus/storefront',
+      },
+    ],
+  },
+  {
+    class: 'AddToCartComponent',
```

Comment:
the previous one is also `AddToCartComponent`

Reply:
Yup, there were 2 constructors in 4.x.  The deprecatedParams signature is different

Reply:
yup I think it's good because it took the changes and generated based on the constructor overloads